### PR TITLE
 update and delete methods for assignments

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
Below are the Postman tests for the delete and update method of assignments:-

1). Update by ID

image

2). Delete by ID

image

### CODE REVIEW

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09  pls see this one

File: src/assignment/assignment.controller.ts [hunk 1]
```diff
@@ -88,8 +87,8 @@ export class AssignmentController {
   }
 
   // Delete a Assignment
-  @Delete('/delete')
-  async deleteAssignment(@Res() res, @Query('uid') uid) {
+  @Delete('/:uid')
```
Comment (Abhishek-kumar09) [inline src/assignment/assignment.controller.ts:90]: Why you have named it uid?
name it something like AssignmentId or so.

File: src/assignment/assignment.controller.ts [hunk 2]
```diff
@@ -67,10 +66,10 @@ export class AssignmentController {
     return res.status(HttpStatus.OK).json(Assignment);
   }
 
-  @Put('/update')
+  @Put('/:uid')
```
Comment (Abhishek-kumar09) [inline src/assignment/assignment.controller.ts:69]: same here

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]

File: src/assignment/assignment.controller.ts (hunk 1 above)
Comment (Devesh21700Kumar) [reply src/assignment/assignment.controller.ts:90]: Ok ,will change this

File: src/assignment/assignment.controller.ts (hunk 2 above)
Comment (Devesh21700Kumar) [reply src/assignment/assignment.controller.ts:69]: Ok ,will change this

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09 I've made the asked changes for this PR.

File: src/modules/assignment/assignment.controller.ts [hunk 3]
```diff
@@ -67,15 +66,15 @@ export class AssignmentController {
     return res.status(HttpStatus.OK).json(Assignment);
   }
 
-  @Put('/update')
+  @Put('/:AssignmentId')
```
Comment (Abhishek-kumar09) [inline src/modules/assignment/assignment.controller.ts:69]: use camel case

File: src/modules/assignment/assignment.controller.ts [hunk 4]
```diff
@@ -88,9 +87,14 @@ export class AssignmentController {
   }
 
   // Delete a Assignment
-  @Delete('/delete')
-  async deleteAssignment(@Res() res, @Query('uid') uid) {
-    const Assignment = await this.AssignmentService.deleteAssignment(uid);
+  @Delete('/:AssignmentId')
```
Comment (Abhishek-kumar09) [inline src/modules/assignment/assignment.controller.ts:90]: use camelcase here

File: src/modules/assignment/assignment.controller.ts [hunk 5]
```diff
@@ -88,9 +87,14 @@ export class AssignmentController {
...
+  async deleteAssignment(
+    @Res() res,
```
Comment (Abhishek-kumar09) [inline src/modules/assignment/assignment.controller.ts:92]: We will not be using `@Res`. return the object as it is.
Like this: https://github.com/codeforcauseorg/edu-server/commit/2e30df8a0300789d2034b7228086a1f3f4529fdb#r49621473

File: src/modules/assignment/assignment.controller.ts [hunk 6]
```diff
@@ -88,9 +87,14 @@ export class AssignmentController {
...
+    @Param('AssignmentId') AssignmentId: string,
+  ) {
+    const Assignment = await this.AssignmentService.deleteAssignment(
+      AssignmentId,
+    );
```
Comment (Abhishek-kumar09) [inline src/modules/assignment/assignment.controller.ts:97]: It can be written in one line

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]
Some nits

File: src/modules/assignment/assignment.controller.ts (hunk 6 above)
Comment (Devesh21700Kumar) [reply src/modules/assignment/assignment.controller.ts:97]: Linting makes this appear in a new line (npm run lint:fix)

File: src/modules/assignment/assignment.controller.ts (hunk 5 above)
Comment (Devesh21700Kumar) [reply src/modules/assignment/assignment.controller.ts:92]: Ok

File: src/modules/assignment/assignment.controller.ts (hunk 4 above)
Comment (Devesh21700Kumar) [reply src/modules/assignment/assignment.controller.ts:90]: assignmentId

File: src/modules/assignment/assignment.controller.ts (hunk 3 above)
Comment (Devesh21700Kumar) [reply src/modules/assignment/assignment.controller.ts:69]: okay

File: src/modules/assignment/assignment.controller.ts (hunk 5 above)
Comment (Devesh21700Kumar) [reply src/modules/assignment/assignment.controller.ts:92]: changes made

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09 

1).assignmentId camelcase
2).removed @Res decorator use

File: src/modules/assignment/assignment.controller.ts [hunk 7]
```diff
@@ -37,38 +35,31 @@ export class AssignmentController {
...
-  ) {
+  async addAssignment(@Body() CreateAssignmentDTO: CreateAssignmentDTO) {
     const Assignment = await this.AssignmentService.addAssignment(
       CreateAssignmentDTO,
     );
-    return res.status(HttpStatus.OK).json({
-      message: 'Assignment has been created successfully',
-      Assignment,
-    });
+    return Assignment;
```
Comment (Abhishek-kumar09) [inline src/modules/assignment/assignment.controller.ts:42]: `Assignment` -> `assignment`

File: src/modules/assignment/assignment.controller.ts [hunk 8]
```diff
@@ -37,38 +35,31 @@ export class AssignmentController {
...
   }
 
   // Retrieve Assignments list
   @ApiCreatedResponse({ type: [AssignmentResponseBody] })
   @Get()
-  async getAllAssignment(@Res() res) {
+  async getAllAssignment() {
     const Assignments = await this.AssignmentService.getAllAssignment();
-    return res.status(HttpStatus.OK).json(Assignments);
+    return Assignments;
```
Comment (Abhishek-kumar09) [inline src/modules/assignment/assignment.controller.ts:50]: const and vars start with small letters

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]

File: src/modules/assignment/assignment.controller.ts (hunk 8 above)
Comment (Devesh21700Kumar) [reply src/modules/assignment/assignment.controller.ts:50]: ok will do

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09 

3). const and vars starting with small letter

File: src/modules/assignment/assignment.controller.ts [hunk 9]
```diff
@@ -38,63 +35,52 @@ export class AssignmentController {
...
+    const assignment = await this.AssignmentService.getAssignment(AssignmentId);
+    return assignment;
   }
 
-  @Put('/update')
+  @Put('/:assignmentId')
   async updateAssignment(
-    @Res() res,
-    @Query('uid') uid,
+    @Param('assignmentId') AssignmentId: string,
```
Comment (Abhishek-kumar09) [inline src/modules/assignment/assignment.controller.ts:63]: `AssignmentId` -> `assignmentId`

File: src/modules/assignment/assignment.controller.ts [hunk 10]
```diff
@@ -38,63 +35,52 @@ export class AssignmentController {
...
-  async deleteAssignment(@Res() res, @Query('uid') uid) {
-    const Assignment = await this.AssignmentService.deleteAssignment(uid);
-    if (!Assignment) throw new NotFoundException('Assignment does not exist');
-    return res.status(HttpStatus.OK).json({
-      message: 'Assignment has been deleted',
-      Assignment: Assignment,
-    });
+  @Delete('/:assignmentId')
+  async deleteAssignment(@Param('assignmentId') AssignmentId: string) {
+    const assignment = await this.AssignmentService.deleteAssignment(
```
Comment (Abhishek-kumar09) [inline src/modules/assignment/assignment.controller.ts:80]: `AssignmentId` -> `assignmentId`

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09 
4).  Assignment  to assignmentId Param 

Also after this, PR.. i think I would also create a separate updateassignment data transfer object for this soon

Would it be fine ?

**Pull request merged**